File: src-tauri/src/types/api/release.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

#[derive(Debug, Serialize, Deserialize)]
pub struct Release {
    pub platforms: HashMap<String, PlatformInfo>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct PlatformInfo {
    pub signature: String,
    pub version: String,
    pub pub_date: String, // 可改为 chrono::DateTime<FixedOffset> 做时间解析
    pub force_update: bool,
    pub notes: Vec<String>,
    pub url: String,
    pub size: u64,
}

File: src/types/release.ts
import { u32 } from "./u32";

interface Release {
  signature: string;
  version: string;
  pub_date: string; // RFC 3339 格式的字符串 "2025-07-02T15:43:00+08:00"
  force_update: boolean;
  notes: string[];
  url: string;
  size: u32;
}

export type { Release };
